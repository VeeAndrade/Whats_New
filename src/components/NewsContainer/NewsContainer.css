.news-container {
  display: grid;
  height: 90vh;
  width: 80%;
  grid-template-columns: 1fr 1fr;
  grid-row-gap: 20px;
  overflow: scroll;
}