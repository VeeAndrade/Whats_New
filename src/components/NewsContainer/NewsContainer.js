import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle.js'


const NewsContainer = (props) => {
  const localNews = props.currentPage.map(article => {
    return <NewsArticle 
    headline={article.headline} 
    image={article.img}
    description={article.description}
    url={article.url}
    key={article.id}
    />
  })
  return (
    <section className='news-container'>
    {localNews}
    </section>
  )
}

export default NewsContainer;