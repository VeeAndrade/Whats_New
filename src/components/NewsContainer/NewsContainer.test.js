import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer'

describe('NewsContainer', () => {
  it('should render all the news articles', () => {
    const wrapper = shallow( <NewsContainer currentPage= {[{
          headline: 'The Underworld',
          image: 'image@url.com',
          description: 'The screams from hell finally come through',
          url: 'url@url.com'
        }]}
        />)

    expect(wrapper).toMatchSnapshot();
  });
});