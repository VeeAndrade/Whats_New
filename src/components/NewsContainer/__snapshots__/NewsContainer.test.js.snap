// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should render all the news articles 1`] = `
<section
  className="news-container"
>
  <NewsArticle
    description="The screams from hell finally come through"
    headline="The Underworld"
    url="url@url.com"
  />
</section>
`;
